homeassistant/components/google/calendar.py

```diff
@@ -37,23 +39,51 @@
     DOMAIN,
     SERVICE_SCAN_CALENDARS,
 )
-from .const import DISCOVER_CALENDAR
+from .const import (
+    CONF_CALENDAR_ACCESS,
+    DISCOVER_CALENDAR,
+    EVENT_DESCRIPTION,
+    EVENT_END_CONF,
+    EVENT_END_DATE,
+    EVENT_END_DATETIME,
+    EVENT_IN,
+    EVENT_IN_DAYS,
+    EVENT_IN_WEEKS,
+    EVENT_START_CONF,
+    EVENT_START_DATE,
+    EVENT_START_DATETIME,
+    EVENT_SUMMARY,
+    EVENT_TYPES_CONF,
+    FeatureAccess,
+)
 
 _LOGGER = logging.getLogger(__name__)
 
-DEFAULT_GOOGLE_SEARCH_PARAMS = {
-    "orderBy": "startTime",
-    "singleEvents": True,
-}
-
 MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=15)
 
 # Events have a transparency that determine whether or not they block time on calendar.
 # When an event is opaque, it means "Show me as busy" which is the default.  Events that
 # are not opaque are ignored by default.
-TRANSPARENCY = "transparency"
 OPAQUE = "opaque"
 
+_EVENT_IN_TYPES = vol.Schema(
+    {
+        vol.Exclusive(EVENT_IN_DAYS, EVENT_TYPES_CONF): cv.positive_int,
+        vol.Exclusive(EVENT_IN_WEEKS, EVENT_TYPES_CONF): cv.positive_int,
+    }
+)
+
+SERVICE_CREATE_EVENT = "create_event"
+CREATE_EVENT_SCHEMA = {
+    vol.Required(EVENT_SUMMARY): cv.string,
+    vol.Optional(EVENT_DESCRIPTION, default=""): cv.string,
+    vol.Exclusive(EVENT_START_DATE, EVENT_START_CONF): cv.date,
+    vol.Exclusive(EVENT_END_DATE, EVENT_END_CONF): cv.date,
+    vol.Exclusive(EVENT_START_DATETIME, EVENT_START_CONF): cv.datetime,
+    vol.Exclusive(EVENT_END_DATETIME, EVENT_END_CONF): cv.datetime,
+    vol.Exclusive(EVENT_IN, EVENT_START_CONF): _EVENT_IN_TYPES,
```

Comment:
Let's add a validator that checks that at least one of the three ways of specifying time is present and that both start and end time arguments are passed when needed.

https://github.com/home-assistant/core/blob/576fc9dc643fdb98be58ca7e3e56e7c3ecc52a02/homeassistant/helpers/config_validation.py#L134

Start and end should probably use `vol.Inclusive`.

Reply:
Updated. I updated the existing event as well to keep the tests simple.